### PR TITLE
cmd/embeddedrules: added check for int needle

### PR DESCRIPTION
**Pull request by i582**

### CODE REVIEW

**Comment by quasilyte** [comment]
Could you add a quickfix test?

File: src/cmd/embeddedrules/rules.php [hunk 1]
```diff
@@ -368,3 +368,26 @@ function indexingSyntax() {
      */
     $x{$y};
 }
+
+/**
+ * @comment Report using an integer for needle.
```
Comment (quasilyte) [inline src/cmd/embeddedrules/rules.php:373]:
```
Report using an integer for needle
```
=>
```
Report using an integer for $needle argument of str* functions
```

**Comment by quasilyte** [comment]
Also, I would add tests with non-precise type arguments.

File: src/cmd/embeddedrules/rules.php (hunk 1 above)
Comment (i582) [reply src/cmd/embeddedrules/rules.php]: Done.

**Comment by i582** [comment]
> Also, I would add tests with imprecise type arguments.

Added.

**Comment by i582** [comment]
> Could you add a quickfix test?

The presence of `@fix` in the rule seems to be a bug.
I don’t think it’s a good idea to add the same description for 
each function in the rule in order to add a fix that is most likely 
not completely safe.

**Review by quasilyte — APPROVED** [review]

**Pull request merged**